### ISSUE TITLE
Are there any strategies available for preserving cell type similarity when conducting cross-species co-embedding?

### ISSUE THREAD

**Issue by lijinbio** [snap-stanford/SATURN#5]
Hello,

I attempted to utilize SATURN for cross-species co-embedding training, but observed that cell type similarity was not maintained after co-embedding. Specifically, cell types that were highly similar in one species became significantly separated in the resulting co-embedding. I am curious about whether it is possible to properly configure SATURN to retain cell type similarity during the co-embedding generation process. Thank you.

**Comment by Yanay1** [comment]
Sorry to clarify: do you mean that similar cell type within the same species are separated in the embedding? 

If that’s the case, you could do a few things:
- double check that this occurs by doing a nearest neighbors style analysis between the cell type, visualizations like UMAP or tSNE can be misleading.
- you could rerun SATURN with those cell types merged
- you could run SATURN with fewer epochs for metric learning

**Comment by lijinbio** [comment]
Thank you, @Yanay1, for your prompt response. I had two similar types within a single species, but they are separated in the embedding. Regarding the nearest neighbor style analysis, I utilized hierarchical clustering using the Euclidean distance between the average latent embeddings of cell types. The resulting clustering is consistent with UMAP, and the two types are significantly distant from each other. While it is true that merged cell types can integrate well with other species, my objective is to preserve the individual cell types to enable label transfer from other species. I am not entirely clear about your third suggestion. Could you elaborate on the epochs for metric learning? Thank you.

**Comment by Yanay1** [comment]
To reduce the number of epochs for metric learning, you could add the `epochs` argument to SATURN and set it to a number less than 50, for example 10. You could also look at the pretraining AnnData produced by SATURN to see if the cell types are separated there.


Can I ask what the two cell types are (and how they were annotated)? Your best option may be to merge the two, or you could also try transferring labels with the existing integration.

**Comment by lijinbio** [comment]
I appreciate your explanation, and I now have a better understanding of epochs. Thank you. I have attached example UMAPs of the pre-trained object that includes BC8 and BC9 cell types from the same species.

<img width="474" alt="Screen Shot 2023-03-15 at 9 27 46 PM" src="https://user-images.githubusercontent.com/16247242/225495609-e8e500a8-a202-4dc6-bdd4-66e4894fd08c.png">

These two cell types are similar in terms of their transcriptomes and are located near each other in the UMAP. They are annotated by existing public cell-type markers. However, the pre-trained object already shows some degree of distance between them.

<img width="667" alt="Screen Shot 2023-03-15 at 9 29 04 PM" src="https://user-images.githubusercontent.com/16247242/225495669-afe0f4e8-6f9b-494c-ace4-34f1c9fcf300.png">

<img width="512" alt="Screen Shot 2023-03-15 at 9 30 01 PM" src="https://user-images.githubusercontent.com/16247242/225495678-a0469d27-8bad-40e6-8407-38d0306e913c.png">

I recognize that the merged cell type has a well-formed embedding. Can you please provide more details about how to transfer labels using the existing integration? Thank you.

**Comment by Yanay1** [comment]
What does the embedding look like after metric learning?

**Comment by lijinbio** [comment]
Below is the embedding after training. (Is this the metric learning you are looking for? I am sorry for not being familiar with the terminology used here. Thanks).

<img width="489" alt="Screen Shot 2023-03-15 at 9 53 17 PM" src="https://user-images.githubusercontent.com/16247242/225498343-8ad965a2-fd58-4f85-85b2-9f7ae55aa028.png">

**Comment by Yanay1** [comment]
It is, thanks!

My suggestions would be:
- Try to run SATURN with BC8 and BC9 merged
- Increase the number of macrogenes from 2000 to 3000 or 4000. The data you're integrating seems to containt very related cell types so they may have very niche transcriptional differences.
- Try to do differential expression in the macrogene space between BC8 and BC9. There may be some additional transcriptional differences between them that are picked up by SATURN. It might also be worth looking at the protein embeddings and top-weighted macrogenes for each of their respective marker genes.
- You can try running SATURN using multiple seeds, and then checking their alignment across multiple seeds.

**Comment by lijinbio** [comment]
Thank you for your valuable suggestions. I appreciate your input, and I will definitely give them a try.
